File: java/src/insert_sort.java
import java.util.*;

public class insert_sort {
    public static int[] InsertSort(int[] list) {
        for (int i = 1; i < list.length; i++) {
            int j = i;
            int cur = list[i];
            while (j > 0 && cur < list[j - 1]) {
                list[j] = list[j - 1];
                j--;
            }
            list[j] = cur;
        }

        return list;
    }

    public static void main(String[] args) {
        int[] scores = { 5, 1, 0, 2, 7, 3 };
        System.out.println(Arrays.toString(InsertSort(scores)));
    }
}

File: c/insert_sort.c
#include <stdio.h>
#include "sort_test.c"

int *insert_sort(int list[], int size)
{
    for (int i = 0; i < size; i++)
    {
        int j = i;
        int cur = list[i];
        while (j > 0 && cur < list[j - 1])
        {
            list[j] = list[j - 1];
            j--;
        }
        list[j] = cur;
    }

    return list;
}

void main()
{
    sort_test(insert_sort);
}